HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+| **Recordings in target JVMs**                                             |                                                                         |
+| Get a list of recordings in a target JVM                                  | [`TargetRecordingsGetHandler`](#TargetRecordingsGetHandler)             |
+| Create a snapshot recording in a target JVM                               | [`TargetSnapshotPostHandler`](#TargetSnapshotPostHandler)               |
+| Start a recording in a target JVM                                         | [`TargetRecordingsPostHandler`](#TargetRecordingsPostHandler)           |
+| Stop a recording in a target JVM                                          | [`TargetRecordingPatchHandler`](#TargetRecordingPatchHandler)           |
+| Delete a recording in a target JVM                                        | [`TargetRecordingDeleteHandler`](#TargetRecordingDeleteHandler)         |
+| Download a recording in a target JVM                                      | [`TargetRecordingGetHandler`](#TargetRecordingGetHandler)               |
+| Download a report of a recording in a target JVM                          | [`TargetReportGetHandler`](#TargetReportGetHandler)                     |
+| Save a recording in a target JVM to persistent storage                    | [`TargetRecordingPatchHandler`](#TargetRecordingPatchHandler)           |
+| Upload a recording from a target JVM to the Grafana datasource            | [`TargetRecordingUploadPostHandler`](#TargetRecordingUploadPostHandler) |
+| **Recordings in persistent storage**                                      |                                                                         |
+| Get a list of recordings in persistent storage                            | [`RecordingsGetHandler`](#RecordingsGetHandler)                         |
+| Upload a recording to persistent storage                                  | [`RecordingsPostHandler`](#RecordingsPostHandler)                       |
+| Delete a recording from persistent storage                                | [`RecordingDeleteHandler`](#RecordingDeleteHandler)                     |
+| Download a recording in persistent storage                                | [`RecordingGetHandler`](#RecordingGetHandler)                           |
+| Download a report of a recording in persistent storage                    | [`ReportGetHandler`](#ReportGetHandler)                                 |
+| Upload a recording from persistent storage to the Grafana datasource      | [`RecordingUploadPostHandler`](#RecordingUploadPostHandler)             |
+
+
+### Core
+
+* #### `AuthPostHandler`
+
+    ###### synopsis
+    Attempts user authenticaiton;
+    used as a simple way to check a user authentication header.
+
+    ###### request
+    `POST /api/v1/auth`
+
+    The request should include an `Authorization` header to be checked.
+
+    ###### response
+    `200` - No body. Getting this response means that the header is valid.
```

Comment:
`that the header is valid and that the user has been successfully authenticated`, maybe?